Mage/src/main/java/mage/abilities/effects/BecomesAttachedToCreatureSourceEffect.java

```diff
@@ -0,0 +1,119 @@
+ * As per rule 614.1c, this should be a replacement effect.
+ * See [[Sanctuary Blade]].
+ *
+ * @author DominionSpy
+ */
+public class BecomesAttachedToCreatureSourceEffect extends ReplacementEffectImpl {
+
+    protected Effects baseEffects = new Effects();
+    protected String text;
+    protected Condition condition;
+    protected boolean optional;
+
+    public static final String SOURCE_OBJECT_STACK_ABILITY = "sourceObjectStackAbility";
+
+    public BecomesAttachedToCreatureSourceEffect(Effect baseEffect) {
+        this(baseEffect, "");
+    }
+
+    public BecomesAttachedToCreatureSourceEffect(Effect baseEffect, String text) {
+        this(baseEffect, text, false);
+    }
+
+    public BecomesAttachedToCreatureSourceEffect(Effect baseEffect, String text, boolean optional) {
+        this(baseEffect, null, text, optional);
+    }
+
+    public BecomesAttachedToCreatureSourceEffect(Effect baseEffect, Condition condition, String text, boolean optional) {
+        super(Duration.WhileOnBattlefield, baseEffect.getOutcome(), false);
+        this.baseEffects.add(baseEffect);
+        this.text = text;
+        this.condition = condition;
+        this.optional = optional;
+    }
+
+    protected BecomesAttachedToCreatureSourceEffect(final BecomesAttachedToCreatureSourceEffect effect) {
+        super(effect);
+        this.baseEffects = effect.baseEffects.copy();
+        this.text = effect.text;
+        this.condition = effect.condition;
+        this.optional = effect.optional;
+    }
+
+    @Override
+    public BecomesAttachedToCreatureSourceEffect copy() {
+        return new BecomesAttachedToCreatureSourceEffect(this);
+    }
+
+    public void addEffect(Effect effect) {
+        baseEffects.add(effect);
+    }
+
+    @Override
+    public boolean checksEventType(GameEvent event, Game game) {
+        return GameEvent.EventType.ATTACH == event.getType();
+    }
+
+    @Override
+    public boolean applies(GameEvent event, Ability source, Game game) {
+        if (event.getSourceId().equals(source.getSourceId())) {
+            return condition == null || condition.apply(game, source);
+        }
+        return false;
+    }
+
+    @Override
+    public boolean replaceEvent(GameEvent event, Ability source, Game game) {
+        if (optional) {
+            Player controller = game.getPlayer(source.getControllerId());
+            MageObject object = game.getPermanentEntering(source.getSourceId());
+            if (object == null) {
+                object = game.getObject(source);
+            }
+            if (controller == null || object == null) {
+                return false;
+            }
+            if (!controller.chooseUse(outcome, "Use effect of " + object.getLogName() + '?', source, game)) {
+                return false;
+            }
+        }
+        StackObject attachesByObject = game.getStack().getStackObject(event.getSourceId());
```

Comment:
Can you explain what this is doing? I'm not convinced it is relevant for attaching.